repository_service_tuf_worker/repository.py

```diff
@@ -398,112 +427,172 @@ def bootstrap(
 
-    def add_targets(self, payload: Dict[str, Any], update_state: str) -> None:
+    def publish_targets(self):
+        """
+        Publish targets from SQL DB as persistent TUF Metadata in the backend
+        storage.
+        """
+
+        # lock to avoid race conditions
+        with self._redis.lock("publish_targets", timeout=5.0):
+            # get all delegated role names with unpublished targets
+            unpublished_roles = targets_crud.read_unpublished_rolenames(
+                self._db
+            )
+
+            if len(unpublished_roles) == 0:
+                logging.debug("No new targets in delegated roles. Finishing")
+                return None
+
+            # initialize the new snapshot targets meta and published targets
+            # from DB SQL
+            new_snapshot_meta: List[Tuple(str, int)] = []
+            db_published_targets: List[targets_models.RSTUFTargets] = []
+            for _, rolename in unpublished_roles:
+                # get the unpublished targets for the delegated, it will be use
+                # to update the database when Snapshot and Timestamp is
+                # published by `_update_timestamp`
+                db_targets = targets_crud.read_unpublished_by_rolename(
+                    db=self._db, rolename=rolename
+                )
+                logging.debug(f"{rolename}: New targets #: {len(db_targets)}")
+                db_published_targets += db_targets
+
+                # load the delegated targets role, clean the targets and add
+                # a new meta from the SQL DB.
+                # note: it might include targets from another parent task, it
+                # will speed up the process of publishing new targets.
+                role = self._load(rolename)
+                role.signed.targets.clear()
+                role.signed.targets = {
+                    target[0]: TargetFile.from_dict(target[1], target[0])
+                    for target in targets_crud.read_all_add_by_rolename(
+                        self._db, rolename
+                    )
+                }
+
+                # update expiry, bump version and persist to the storage
+                self._bump_expiry(role, BINS)
+                self._bump_version(role)
+                self._sign(role, BINS)
+                self._persist(role, rolename)
+                # append to the new snapshot targets meta
+                new_snapshot_meta.append((rolename, role.signed.version))
+
+            # update snapshop and timestamp
+            # note: the `db_published_targes` contains the targets that
+            # needs to updated in SQL DB as 'published' and it will be done
+            # by the `_update_timestamp`
+            self._update_timestamp(
+                self._update_snapshot(new_snapshot_meta),
+                db_published_targets,
+            )
+
+    def add_targets(
+        self, payload: Dict[str, Any], update_state: Task.update_state
+    ) -> Dict[str, Any]:
```

Comment:
It seems the docstring you added for `payload` and `update_state` has been removed somewhere.
Could you add it again?
We discussed this in https://github.com/vmware/repository-service-tuf-worker/pull/145#discussion_r1056417965,

Reply:
As you asked to add docstring because of the wrong typing for `update_state`,  I fixed the correct typing, which makes more clear.